packages/docs/src/guide.md

```diff
@@ -29,16 +32,20 @@ export default {
 
 Vuelidate comes with a set of validators, that live inside the `@vuelidate/validations` package, which you must install separately. For a full list of available validators, check the [Validators](./validators.md) page.
 
-Validators can be as simple as functions returning a boolean, you are free to write your own or use third party validators. Read more about how to write your own validators on the [Custom Validators](./custom_validators.md) page.
+Validators are functions that return a boolean value after checking the validity of the state of a form.
+
+Vuelidate also supports writing your own custom validators, you can learn more about how to write your own validators on the [Custom Validators](./custom_validators.md) page.
 
 ### Checking for validation state
 
-We have a our validation rules ready, now we need to check for errors.
+Now that we have our validation rules set up, we can start checking for errors.
 
-Vuelidate builds a validation state, that can be accessed via the `v$` property. It is a nested object that roughly follows your `validations` structure,
-but with some extra validity related properties. It is reactive, meaning it changes as you type.
+Vuelidate builds a validation state, that can be accessed via the `v$` property. It is a nested object that follows your `validations` structure,
+but with some extra validity related properties.
 
-So for our form above, to check whether name is valid, we would can see if the `name.$error` property is `true` or not:
+`v$` is also reactive - meaning it changes as the user types.
+
+Building up on our form example above, to check whether name is valid we can now check to see if the `name.$error` property is `true` or `false`, and display an error for our users.
```

Comment:
I think we have omitted the fact that now all validators are lazy by default. This means that even if someone types anything, the `$invalid` and `$error` state will still be `false` because the validators won’t even run until a property is `$dirty.`

I imagine this might cause confusion as the examples presented just won’t work.

Reply:
Can you provide an example of how to correctly handle this? Are we suggesting they check for both $dirty and $error? Will this be addressed if we check for $errors.length?

Reply:
In the previous vuelidate version all validations happened instantly and the `$invalid` state was always up to date. You then had to call `$touch` to change the `$dirty` state. `$error` was just a logical `$invalid && $dirty` right?

Now, this still works similarly for `$error` and `$errors`. The difference is that all validators are lazy by default and are not executed unless the `$dirty` state is changed to `true`. Thus until someone uses `$touch()` or `$model` or passes `$autoDirty: true`, there is no validation being done at all. @dobromir-hristov also recently introduced:
```
this.v$.$validate().then((result) => {
  console.log('Result is', result)
})
```
Which internally triggers `$touch()` and sets `$dirty` state to true. So there are 4 ways to start validating.

This kind of brings the questions – should we even have `$invalid` at this point for any other reason that backwards compatibility? That’s because once you call `$reset()` to reset the `$dirty` state, the validators will again stop being executed.